### PR TITLE
BAN-2811: Unify multiply architecture to support multiple tokens

### PR DESCRIPTION
**Pull request by sablevsky**
## Issue link

https://linear.app/banx-gg/issue/BAN-2811/fe-implement-jlp-multiply

## Vercel preview

https://banx-ui-git-feature-ban-2811-frakt.vercel.app/

### CODE REVIEW

File: src/pages/tokenLending/LeveragePage/hooks.ts [hunk 1]
```diff
@@ -236,20 +247,25 @@ export const useLrtsLeverage = () => {
...
   }
 }
 
-export const useLrtsMintConversionRate = (connection: web3.Connection) => {
+export const useCollateralConversionRate = (mint: web3.PublicKey, connection: web3.Connection) => {
+  const queryFn = useMemo(() => {
+    if (mint === lrtsSOL.LRTS_MINT) return lrtsSOL.getSolayerConversionRate
+    return () => Promise.resolve(0)
+  }, [mint])
+
```
Comment (iamsphere) [inline src/pages/tokenLending/LeveragePage/hooks.ts:263]: Смущает неявная передача connection внутрь lrtsSOL.getSolayerConversionRate
Не сразу очевидно что она зависит от connection

const createQueryFn = (mint, connection) => ... func(connection) ?

File: src/pages/tokenLending/LeveragePage/hooks.ts [hunk 2]
```diff
@@ -205,18 +214,20 @@ export const useLrtsLeverage = () => {
...
     )
 
-    if (totalBorrowAmount.gt(LOAN_VALUE_LIMIT))
+    if (pair.loanValueLimit && totalBorrowAmount.gt(pair.loanValueLimit))
       return {
         disabled: true,
-        text: '100◎ limit reached',
+        text: `${pair.loanValueLimit.div(
+          new BN(getTokenDecimals(pair.marketTokenType)),
+        )} ${getTokenUnit(pair.marketTokenType)} limit reached`,
```
[the remark this answers is not in the document]
Comment (sablevsky) [reply src/pages/tokenLending/LeveragePage/hooks.ts]: https://github.com/frakt-solana/banx-ui/pull/1088/commits/abd65dfbcc5d564e9fa31a3f1fec754b133025f8

File: src/pages/tokenLending/LeveragePage/hooks.ts (hunk 1 above)
Comment (sablevsky) [reply src/pages/tokenLending/LeveragePage/hooks.ts]: https://github.com/frakt-solana/banx-ui/pull/1088/commits/b7955a89470415028866ed1efe4abe735b8785f5

File: src/pages/tokenLending/LeveragePage/hooks.ts [hunk 3]
```diff
@@ -236,20 +247,25 @@ export const useLrtsLeverage = () => {
...
+export const useCollateralConversionRate = (mint: web3.PublicKey, connection: web3.Connection) => {
+  const queryFn = useMemo(() => {
+    if (mint === lrtsSOL.LRTS_MINT) return lrtsSOL.getSolayerConversionRate
+    return () => Promise.resolve(0)
+  }, [mint])
+
   const { data: rate, isLoading } = useQuery(
-    [`sSolConversionRate`],
-    () => lrtsSOL.getSolayerConversionRate(connection),
+    ['sSolConversionRate', mint],
```
[the remark this answers is not in the document]
Comment (sablevsky) [reply src/pages/tokenLending/LeveragePage/hooks.ts]: https://github.com/frakt-solana/banx-ui/pull/1088/commits/b7955a89470415028866ed1efe4abe735b8785f5

File: src/pages/tokenLending/LeveragePage/hooks.ts [hunk 4]
```diff
@@ -26,25 +26,29 @@ import {
...
+export const useLrtsLeverage = ({ pair, collateralToken }: UseLrtsLeverageParams) => {
   const wallet = useWallet()
   const { connection } = useConnection()
 
-  const { walletCollateralBalance } = useWalletCollateralBalance(
-    new web3.PublicKey(LRTS_TOKEN_META.mint),
-  )
-  const { rate: lrtsConversionRate, isLoading: isLrtsConversionRateLoading } =
-    useLrtsMintConversionRate(connection)
+  const { walletCollateralBalance } = useWalletCollateralBalance(pair.collateralMint)
```
[the remark this answers is not in the document]
Comment (sablevsky) [reply src/pages/tokenLending/LeveragePage/hooks.ts:48]: https://github.com/frakt-solana/banx-ui/pull/1088/commits/3ded6570c17556c59b1f4bac1511664ab477a4f8

**Review by iamsphere — APPROVED** [review]

**Pull request merged**